diff -ur libssh-0.6.4/cmake/Modules/FindOpenSSL.cmake libssh-0.6.4.orig/cmake/Modules/FindOpenSSL.cmake
--- libssh-0.6.4/cmake/Modules/FindOpenSSL.cmake	2014-09-16 00:24:34.000000000 +0530
+++ libssh-0.6.4.orig/cmake/Modules/FindOpenSSL.cmake	2014-12-30 16:29:50.895051986 +0530
@@ -138,22 +138,32 @@
             mark_as_advanced(LIB_EAY_DEBUG LIB_EAY_RELEASE)
         elseif (MINGW)
             # same player, for MingW
-            find_library(LIB_EAY
+            find_library(OPENSSL_SSL_LIBRARIES
                 NAMES
-                    libeay32
+                    ssl
+                    ssleay32
+                    ssleay32MD
                 PATHS
-                    ${OPENSSL_ROOT_DIR}/lib/MinGW
+                    ${_OPENSSL_LIBDIR}
+                    /opt/local/lib
+                    /sw/lib
+                    /usr/sfw/lib/64
+                    /usr/sfw/lib
             )
 
-            find_library(SSL_EAY
+            find_library(OPENSSL_CRYPTO_LIBRARIES
                 NAMES
-                    ssleay32
+                    crypto
                 PATHS
-                    ${OPENSSL_ROOT_DIR}/lib/MinGW
+                    ${_OPENSSL_LIBDIR}
+                    /opt/local/lib
+                    /sw/lib
+                    /usr/sfw/lib/64
+                    /usr/sfw/lib
             )
 
-            mark_as_advanced(SSL_EAY LIB_EAY)
-            set(OPENSSL_LIBRARIES ${SSL_EAY} ${LIB_EAY})
+            mark_as_advanced(OPENSSL_CRYPTO_LIBRARIES OPENSSL_SSL_LIBRARIES)
+            set(OPENSSL_LIBRARIES ${OPENSSL_SSL_LIBRARIES} ${OPENSSL_CRYPTO_LIBRARIES})
         else(MSVC)
             # Not sure what to pick for -say- intel, let's use the toplevel ones and hope someone report issues:
             find_library(LIB_EAY
@@ -174,32 +184,7 @@
             set(OPENSSL_LIBRARIES ${SSL_EAY} ${LIB_EAY})
         endif(MSVC)
     else (WIN32 AND NOT CYGWIN)
-        find_library(OPENSSL_SSL_LIBRARIES
-            NAMES
-                ssl
-                ssleay32
-                ssleay32MD
-            PATHS
-                ${_OPENSSL_LIBDIR}
-                /opt/local/lib
-                /sw/lib
-                /usr/sfw/lib/64
-                /usr/sfw/lib
-        )
-
-        find_library(OPENSSL_CRYPTO_LIBRARIES
-            NAMES
-                crypto
-            PATHS
-                ${_OPENSSL_LIBDIR}
-                /opt/local/lib
-                /sw/lib
-                /usr/sfw/lib/64
-                /usr/sfw/lib
-        )
 
-        mark_as_advanced(OPENSSL_CRYPTO_LIBRARIES OPENSSL_SSL_LIBRARIES)
-        set(OPENSSL_LIBRARIES ${OPENSSL_SSL_LIBRARIES} ${OPENSSL_CRYPTO_LIBRARIES})
     endif (WIN32 AND NOT CYGWIN)
 
     include(FindPackageHandleStandardArgs)
diff -ur libssh-0.6.4/src/CMakeLists.txt libssh-0.6.4.orig/src/CMakeLists.txt
--- libssh-0.6.4/src/CMakeLists.txt	2014-12-19 13:41:17.000000000 +0530
+++ libssh-0.6.4.orig/src/CMakeLists.txt	2014-12-30 17:18:25.263160794 +0530
@@ -288,6 +288,3 @@
   )
 endif (WITH_STATIC_LIB)
 
-if (Threads_FOUND)
-    add_subdirectory(threads)
-endif (Threads_FOUND)
